src/app/page/component/page/cyoa-card-collection-page/cyoa-card-collection-page.component.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { ActivatedRoute, ParamMap } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { BehaviorSubject } from 'rxjs';
+import { PageService } from '../../../service/page-service.service';
+import { CYOACardCollectionComponent } from './cyoa-card-collection-page.component';
+
+describe('CYOACardCollectionComponent', () => {
+  let component: CYOACardCollectionComponent;
+  let fixture: ComponentFixture<CYOACardCollectionComponent>;
+  let setCardUrlSpy: jasmine.Spy;
+
+  const paramMapSubject = new BehaviorSubject<ParamMap>({
+    get: (_: string) => '0'
+  } as ParamMap);
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [CYOACardCollectionComponent],
+      imports: [RouterTestingModule],
+      providers: [
+        {
+          provide: ActivatedRoute,
+          useValue: {
+            paramMap: paramMapSubject.asObservable()
+          }
+        },
+        PageService
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CYOACardCollectionComponent);
+    component = fixture.componentInstance;
+    component.setCardUrl = (card: number) => card;
+    setCardUrlSpy = spyOn(component, 'setCardUrl').and.callThrough();
```

Comment:
I think that if you remove `.and.callThrough()`, the original method isn't called, so you wouldn't have to overwrite it.

```suggestion
    setCardUrlSpy = spyOn(component, 'setCardUrl');
```

Reply:
I get an error when doing that. `Error: <spyOn> : setCardUrl() method does not exist`

Reply:
So sorry, I didn't realize this was the other component and that method is an input.